### PR TITLE
Provide ability to build without publish #7

### PR DESCRIPTION
**Pull request by tjtelan**
Updating README.md with build only PR example
Adding test w/ (respect to upstream) for BUILD_ONLY: true

### CODE REVIEW

**Comment by shalzz** [comment]
Hi, thanks for creating a PR.

Does this work with an actual PR though? I see expect the same errors occurring as the one with the Build tests for this PR because the entrypoint.sh script expects to have the `GITHUB_TOKEN` set but github doesn't set any SECRETS during a PR.

One way to solve this would be to disable the check at https://github.com/shalzz/zola-deploy-action/blob/master/entrypoint.sh#L17 when `BUILD_ONLY` is true?

File: entrypoint.sh [hunk 1]
```diff
@@ -44,18 +48,23 @@ main() {
     echo Building with flags: ${BUILD_FLAGS:+"$BUILD_FLAGS"}
     zola build ${BUILD_FLAGS:+"$BUILD_FLAGS"}
 
-    echo "Pushing artifacts to ${GITHUB_REPOSITORY}:$remote_branch"
+    if ! ${BUILD_ONLY}; then
```
Comment (shalzz) [inline entrypoint.sh:51]: Would you mind switching the logic check here to not have the `not (!)` operator here and switch the resulting code paths.
It'll justify having the `BUILD_ONLY` variable name and make it a bit easier to read.

Comment (tjtelan) [reply entrypoint.sh:51]: Sure, I can make this change right now.

Also, prior to this PR, I tested this change against my own Github Pages site so I could see the results from a PR and a commit to my default branch.

PR: https://github.com/tjtelan/tjtelan.github.io/actions/runs/132888874
Default branch: https://github.com/tjtelan/tjtelan.github.io/actions/runs/132889356

**Comment by shalzz** [comment]
Merged, Thanks!

**Pull request merged**